apps/backend/src/tenable/tenable.controller.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  Controller,
+  Req,
+  Res,
+  Post,
+  Body,
+  HttpException,
+  HttpStatus,
+  All
+} from '@nestjs/common';
+import {TenableService} from './tenable.service';
+import axios from 'axios';
+import {Request, Response} from 'express';
+
+// Extend express-session types to include 'tenable'
+declare module 'express-session' {
+  interface SessionData {
+    tenable?: {
+      host_url: string;
+      accesskey: string;
+      secretkey: string;
+    };
+  }
+}
+
+// NestJS controller that handles Tenable authentication and proxying requests to Tenable
+// It allows users to log in with their Tenable credentials and then proxies all subsequent requests
+// to the Tenable API, handling authentication via session storage.
+@Controller('api/tenable')
+export class TenableController {
+  constructor(private readonly tenableService: TenableService) {}
+
+  @Post('login')
+  /**
+   * Authenticates a user with the provided Tenable credentials and stores them in the session.
+   *
+   * @param req - The incoming HTTP request object, used to access the session.
+   * @param body - An object containing the Tenable host URL, access key, and secret key.
+   * @returns An object indicating success and the authenticated user's data from Tenable.
+   * @throws {HttpException} If any credentials are missing or if authentication fails.
+   */
+  async login(
+    @Req() req: Request,
+    @Body() body: {host_url: string; accesskey: string; secretkey: string}
+  ) {
+    const {host_url, accesskey, secretkey} = body;
+
+    if (!host_url || !accesskey || !secretkey) {
+      throw new HttpException('Missing credentials', HttpStatus.BAD_REQUEST);
```

Comment:
```suggestion
      throw new HttpException('Missing credentials. A host URL, access key, and secret key are required.', HttpStatus.BAD_REQUEST);
```

Do you want to provide more information here? I am not sure what gets printed out, so maybe it is not necessary. I will check in a test and update this comment appropriately.

Reply:
No need to add additional content here, what is sent to the user is set in the frontend